Nuke.Common 10.0.0 can't be build when using Nuke.Tooling.Generator

### Usage Information

Nuke 10.0.0 

### Description

After upgrading to Nuke `10.0.0`, our build fails during code generation for custom tooling (e.g., Yarn, Docker, Kubernetes). The failure occurs because an expected assembly (Nuke.MSBuildTasks.dll) is missing, causing MSBuild to abort during the CodeGenerationTask step.



### Reproduction Steps

1 Use NUKE tooling generation (for some tool Yarn/Docker/Kubernetes or similar) in a project.
2 Upgrade NUKE packages to version `10.0.0`.
3 Run the build using .NET SDK `10.0.100`.
4 Observe MSBuild failing due to the missing Nuke.MSBuildTasks.dll.

### Expected Behavior

Build should not fail and generate the tools.

### Actual Behavior

The build fails with the following MSBuild error:
```
Microsoft (R) .NET SDK version 10.0.100
  Determining projects to restore...
  All projects are up-to-date for restore.
/root/.nuget/packages/nuke.common/10.0.0/build/netcore/Nuke.MSBuildTasks.targets(50,5): error MSB4062: The "Nuke.MSBuildTasks.CodeGenerationTask" task could not be loaded from the assembly /root/.nuget/packages/nuke.common/10.0.0/build/netcore//Nuke.MSBuildTasks.dll. Could not load file or assembly '/root/.nuget/packages/nuke.common/10.0.0/build/netcore/Nuke.MSBuildTasks.dll'. The system cannot find the file specified. [/azp/_work/4/s/src/Nuke/Nuke.csproj]
/root/.nuget/packages/nuke.common/10.0.0/build/netcore/Nuke.MSBuildTasks.targets(50,5): error MSB4062:  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [/azp/_work/4/s/src/Nuke/Nuke.csproj]

Build FAILED.
```

### Regression?

This worked fine with `9.0.4`

### Known Workarounds

It works when I set the TaskEnabled flag to false in the .csproj
```xml
<NukeTasksEnabled>False</NukeTasksEnabled>
```

However this also disables the tool generations, which I dont want.

### Could you help with a pull-request?

No

--- a/source/Nuke.Common/Nuke.Common.csproj
+++ b/source/Nuke.Common/Nuke.Common.csproj
@@ -35,7 +35,7 @@
     <None Include="$(MSBuildProjectName).targets" PackagePath="build" Pack="true" />
     <None Include="..\Nuke.MSBuildTasks\Nuke.MSBuildTasks.targets" PackagePath="build\netcore" Pack="true" />
     <None Include="..\Nuke.MSBuildTasks\Nuke.MSBuildTasks.targets" PackagePath="build\netfx" Pack="true" />
-    <None Include="..\Nuke.MSBuildTasks\bin\$(Configuration)\net8.0\publish\**\*.*" PackagePath="build\netcore" Pack="true" />
+    <None Include="..\Nuke.MSBuildTasks\bin\$(Configuration)\net10.0\publish\**\*.*" PackagePath="build\netcore" Pack="true" />
     <None Include="..\Nuke.MSBuildTasks\bin\$(Configuration)\net472\publish\**\*.*" PackagePath="build\netfx" Pack="true" />
     <None Include="..\Nuke.SourceGenerators\bin\$(Configuration)\netstandard2.0\*.dll" PackagePath="analyzers\dotnet\cs" Pack="true" />
   </ItemGroup>

